buildSrc/src/main/groovy/org/gradle/plugins/classycle/ClassyclePlugin.groovy

```diff
@@ -42,7 +42,7 @@ class ClassyclePlugin implements Plugin<Project> {
             def taskName = sourceSet.getTaskName('classycle', null)
             project.tasks.create(taskName, Classycle, { Classycle task ->
                 task.reportName = sourceSet.name
-                task.classesDir = sourceSet.output.classesDir
+                task.classesDirs = project.files { sourceSet.output.classesDirs.files.findAll { it.exists() } }
```

Comment:
As @oehme stated, this should happen when the task executes. I don't even know why we need the empty directory handling at all. But if we want to do it, it should happen when the task executes in its task action.